MathLib/MathTools.h

```diff
@@ -120,6 +120,16 @@ inline double to_radians(double degrees) {
     return degrees*boost::math::constants::pi<double>()/180.;
 }
 
+template<typename Type> Type limitValueInInterval(const Type variable,
+                                               const Type lower_bound,
+                                               const Type upper_bound)
```

Comment:
Better to pass the types by universal reference (Type&&) to avoid copies if the types are expensive to copy.